server/sonar-server/src/main/java/org/sonar/server/computation/measure/MeasureRepository.java

```diff
@@ -19,17 +19,39 @@
  */
 package org.sonar.server.computation.measure;
 
-import org.sonar.api.measures.Metric;
-import org.sonar.batch.protocol.output.BatchReport;
-
 import com.google.common.base.Optional;
-import org.sonar.core.measure.db.MeasureDto;
+import java.util.Map;
 import org.sonar.server.computation.component.Component;
+import org.sonar.server.computation.metric.Metric;
+import org.sonar.server.computation.metric.MetricImpl;
 
 public interface MeasureRepository {
-  // FIXME should not expose MeasureDto from DAO layer
-  Optional<MeasureDto> findPrevious(Component component, Metric<?> metric);
 
-  // FIXME should not expose Measure from BatchReport
-  Optional<BatchReport.Measure> findCurrent(Component component, Metric<?> metric);
+  /**
+   * Retrieves the base measure (ie. the one currently existing in DB) for the specified {@link Component} for
+   * the specified {@link MetricImpl} if it exists.
+   *
+   * @throws NullPointerException if either argument is {@code null}
+   */
+  Optional<Measure> getBaseMeasure(Component component, Metric metric);
+
+  /**
+   * Retrieves the measure created during the current analysis for the specified {@link Component} for the specified
+   * {@link MetricImpl} if it exists (ie. one created by the Compute Engine or the Batch).
+   */
+  Optional<Measure> getRawMeasure(Component component, Metric metric);
```

Comment:
given the documentation of this method, yes, raw could be dropped

but when you suggested to use "base" and "raw", this paradigms seemed to be used in other places

we should just be consistent